### ISSUE TITLE
Question: matrix-free cell based loop with neighboring data

### ISSUE THREAD

**Issue by btemuer** [exadg/exadg#403]
Hey, I wanted to ask if there have been any developments regarding the matrix-free cell-based face loops in deal.II that we should use in ExaDG, since the comment from below (like several other similar comments) is pretty old, in this case from 4 years ago.

If yes, we could start with removing and replacing this code.

I became aware of this issue after working on the variable coefficient Laplace operator and encountering problems with the face id etc.

https://github.com/exadg/exadg/blob/9f7a5e9bef38ca8e61f1d276d7c27836ceb23ec8/include/exadg/convection_diffusion/spatial_discretization/operators/convective_operator.cpp#L174-L175

**Comment by btemuer** [comment]
Okay, I changed the issue title accordingly.

**Comment by nfehn** [comment]
I do not fully understand the title, because we want to support cell-based face loops in the future, more than we did so far.

**Comment by btemuer** [comment]
You're right. The intention is to only remove the `do_*_integral_cell_based()` kind of functions, as they have been workarounds around the fact that neighboring data was not accessible. These functions are also commented as such in the code. I'm changing the title. Although I'm not sure how to deal with variable coefficients in the case that they are removed..

**Comment by nfehn** [comment]
I think our strategy should be to enable cell-based face loops functionality. On the way, we can remove the code that has become superfluous due to progress in deal.II.

**Comment by nfehn** [comment]
in this context, also note issue #290
